xdsl/transforms/loop_invariant_code_motion.py

```diff
@@ -32,9 +41,66 @@ def can_be_hoisted(op: Operation, target_region: Region) -> bool | None:
     for child in op.walk():
         for operand in child.operands:
             operand_owner = operand.owner
-            assert operand_owner is not None
             if op.is_ancestor(operand_owner):
                 continue
             if target_region.is_ancestor(operand_owner):
                 return False
     return True
+
+
+def _move_loop_invariant_code(region: Region, builder: Builder):
+    # add top-level operations in the loop body to the worklist
+    worklist = deque(op for block in region.blocks for op in block.ops)
```

Comment:
I just thought the `Worklist` class might be better as it avoids duplicates as far as I know, but this probably doesn't matter much.